### PR TITLE
Temporarily adding fluent config generator code

### PR DESCRIPTION
**Pull request by PettitWesley**
### Summary
Add Fluentd/Fluent Bit config generation code. Once we get a separate repo, this will be moved. But to keep us moving forward, we're going to code review it here first.

### Testing
(Just ran the unit tests for this because laziness)
```
$ go test -v -cover ./agent/fluentconfigurationgenerator/...
=== RUN   TestGenerateConfig
--- PASS: TestGenerateConfig (0.00s)
PASS
coverage: 92.9% of statements
ok  	github.com/aws/amazon-ecs-agent/agent/fluentconfigurationgenerator/generator	0.800s	coverage: 92.9% of statements
```

Manual Tests:
- [ ] Test Fluent Bit Generated Config with actual fluent bit to verify the syntax is correct
- [ ] Test Fluentd Generated Config with actual fluentd to verify the syntax is correct

New tests cover the changes: yes

### Description for the changelog
N/A

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by petderek** [comment]
Could you add this to the misc folder?

File: agent/fluentconfigurationgenerator/generator/generator.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package generator
+
+import (
+	"io"
+	"text/template"
+)
+
+type IncludePosition int
```
Comment (fenxiong) [inline agent/fluentconfigurationgenerator/generator/generator.go]: exported names should have comments (https://github.com/golang/go/wiki/CodeReviewComments#doc-comments)

File: agent/fluentconfigurationgenerator/generator/generator_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	}).AddOutput("firehose", "*", map[string]string{
+		"delivery_stream": "my-stream",
+		"region":          "us-west-2",
+	})
+
+	fluentbitConfig := new(bytes.Buffer)
+	config.WriteFluentBitConfig(fluentbitConfig)
+	assert.Equal(t, expectedFluentBitConfig, fluentbitConfig.String(), "Expected Fluent Bit Config to match")
+
+	//config.WriteFluentdConfig(os.Stdout)
```
Comment (fenxiong) [inline agent/fluentconfigurationgenerator/generator/generator_test.go]: should remove?

File: agent/fluentconfigurationgenerator/generator/generator.go (hunk 1 above)
Comment (PettitWesley) [reply agent/fluentconfigurationgenerator/generator/generator.go]: Oh yeah... I meant to do this but then forgot 👍🏻

**Pull request closed without merging**